Packages/com.reese.nav/Agent/Systems/NavCollisionSystem.cs

```diff
@@ -0,0 +1,92 @@
+﻿using Unity.Entities;
+using Unity.Mathematics;
+using Unity.Physics;
+using Unity.Physics.Systems;
+using Unity.Transforms;
+using UnityEngine;
+using Collider = UnityEngine.Collider;
+using RaycastHit = Unity.Physics.RaycastHit;
+
+namespace Reese.Nav
+{
+    /// <summary>
+    /// Projects raycasts infront of any enity which is flocking & walking and adds a new steering force
+    /// whenever those raycasts hit an `obstacle`. It is supposed to 'gently' steer away from obstacles, it does not guarantee anti-clipping.  
+    /// </summary>
+    [UpdateInGroup(typeof(FixedStepSimulationSystemGroup))]
+    [UpdateBefore(typeof(BuildPhysicsWorld))]
+    [UpdateAfter(typeof(NavFlockingSystem))]
+    public class NavCollisionSystem : SystemBase
+    {
+        public bool IsDebugging = true;
+        NavSystem navSystem => World.GetOrCreateSystem<NavSystem>();
+        BuildPhysicsWorld buildPhysicsWorld => World.GetExistingSystem<BuildPhysicsWorld>();
+        protected override void OnUpdate()
+        {
+            var physicsWorld     = buildPhysicsWorld.PhysicsWorld;
+            var settings         = navSystem.Settings;
+            var flockingSettings = navSystem.FlockingSettings;
+            var isDebugging      = IsDebugging;
+
+            Entities
+                .WithNone<NavProblem>()
+                .WithNone<NavPlanning, NavJumping, NavFalling>()
+                .WithAll<NavWalking, NavFlocking, LocalToParent>()
+                .WithReadOnly(physicsWorld)
+                .ForEach((Entity entity, int entityInQueryIndex, ref Translation translation, ref NavAgent agent, ref NavSteering steering,
+                    in LocalToWorld localToWorld, in Rotation rotation, in Parent surface) =>
+                {
+                    var averageHitDirection = float3.zero;
+                    var clostestHitDistance = agent.ObstacleAversionDistance;
+                    int hits = 0;
+                    
+                    // Cast 17 rays in a (flockingSettings.CollisionCastingAngle * 2) deg angle with the entities forward direction
+                    for (int i = 0; i < 17; ++i)
```

Comment:
Maybe the `17` should be defined as a constant, and the comment could be moved above that constant.